runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/EngineConfiguration.java

```diff
@@ -70,6 +76,19 @@ public class EngineConfiguration extends Configuration
     static
     {
         final ConfigurationDef config = new ConfigurationDef("zilla.engine");
+        BiFunction<Configuration, String, URL> decodeFunction = (c, i) ->
+        {
+            try
+            {
+                return URI.create(i).toURL();
+            }
+            catch (MalformedURLException ex)
+            {
+                LangUtil.rethrowUnchecked(ex);
+                return null;
+            }
+        };
+        ENGINE_CONFIG_URL = config.property(URL.class, "config.url", decodeFunction, "file:zilla.json");
```

Comment:
Add a private static method called `configURL` taking `Configuration` and `String` parameters, with implementation same as anonymous `decodeFunction` above, then pass `EngineConfiguration::configURL` instead of `decodeFunction`.

See `KafkaConfiguration.cacheDirectory(Configuration, String)` as an example.